### PR TITLE
k-LIME genmodel

### PR DESCRIPTION
**Pull request by navdeep-G**

### CODE REVIEW

File: h2o-genmodel-extensions/klime/src/main/java/hex/genmodel/algos/klime/EasyPredictKLimeWrapper.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package hex.genmodel.algos.klime;
+
+import hex.ModelCategory;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.exception.PredictException;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.GenModel;
+
+public class EasyPredictKLimeWrapper extends EasyPredictModelWrapper {
```
Comment (michalkurka) [inline h2o-genmodel-extensions/klime/src/main/java/hex/genmodel/algos/klime/EasyPredictKLimeWrapper.java:9]: is this needed for anything?

File: gradle/components/klime-genmodel.gradle [hunk 2]
```diff
@@ -0,0 +1,3 @@
+dependencies {
```
Comment (michalkurka) [inline gradle/components/klime-genmodel.gradle:1]: should instead be listed in the main build.gradle

Comment (michalkurka) [inline gradle/components/klime-genmodel.gradle:1]: imho should also be included in main build.gradle (published projects, java projects...)

File: h2o-genmodel-extensions/klime/src/main/java/hex/genmodel/algos/klime/EasyPredictKLimeWrapper.java (hunk 1 above)
Comment (navdeep-G) [reply h2o-genmodel-extensions/klime/src/main/java/hex/genmodel/algos/klime/EasyPredictKLimeWrapper.java:9]: It is used in k-LIME MOJO tests:
```
 EasyPredictKLimeWrapper wrapper = new EasyPredictKLimeWrapper(_mojo);
```

File: gradle/components/klime-genmodel.gradle (hunk 2 above)
Comment (navdeep-G) [reply gradle/components/klime-genmodel.gradle:1]: What should be included exactly? I followed the same pattern as the xgboost genmodel extension.

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java [hunk 3]
```diff
@@ -791,6 +791,18 @@ public RegressionModelPrediction predictRegression(RowData data, double offset)
     return p;
   }
 
+  public KLimeModelPrediction predictKLime(RowData data) throws PredictException {
```
Comment (michalkurka) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java:794]: this should also be added to the predict() method in this class

Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java:795]: done: https://github.com/h2oai/h2o-3/pull/4618/commits/e1f353f5837fa2c12a2049d0b063d8b87d58f7a4

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java [hunk 4]
```diff
@@ -791,6 +792,18 @@ public RegressionModelPrediction predictRegression(RowData data, double offset)
...
+    double[] preds = preamble(ModelCategory.Regression, data);
```
Comment (navdeep-G) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java:796]: Shouldn't I change this to `preamble(ModelCategory.KLime, data);` @michalkurka ?

Comment (michalkurka) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java:796]: I think that makes sense

Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java]: great, just changed. thx!

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java [hunk 5]
```diff
@@ -791,6 +792,18 @@ public RegressionModelPrediction predictRegression(RowData data, double offset)
     return p;
   }
 
+  public KLimeModelPrediction predictKLime(RowData data) throws PredictException {
+    double[] preds = preamble(ModelCategory.KLime, data);
```
Comment (mmalohlava) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java:796]: I think this expect to that KLime overrides `getModelCategories` - here we pass `ModelCategory.KLime` which is checked against `KLime#getModelCategories` which returns `[Regression]`.

CC: @navdeep-G

**Comment by mmalohlava** [comment]
@michalkurka @Pscheidl @navdeep-G is modification of https://github.com/h2oai/h2o-3/blob/master/h2o-genmodel/src/main/java/hex/genmodel/ModelMojoFactory.java missing here?

**Pull request merged**